doc/espnet2_tutorial.md

```diff
@@ -346,3 +346,243 @@ To enable online decoding, the argument `--use_streaming true` should be added t
+      linear_size: 1024
+      hidden_size: 256
+      heads: 8
+      dropout_rate: 0.1
+      pos_wise_dropout_rate: 0.1
+      att_dropout_rate: 0.1
+      conv_mod_kernel_size: 31
+      num_blocks: 14
+```
+
+#### Decoder
+
+The type can be defined through `decoder` parameter by passing a string (either `rnn` or `stateless`) and the internal parts can be configured
+
+For the decoder, two types of blocks are available: RNN and stateless (only embedding). Contrary to the encoder, the parameters are shared accross the blocks, meaning we only define define only one block here.
+The type of the stack of blocks is by passing a string (either `rnn` or `stateless`) to the parameter `decoder`. The internal parts are defined by the config `decoder_conf` containing the following (optional) parameters:
+
+    decoder_conf:
+      rnn_type (RNN only, optional): Type of RNN cells (int, default = "lstm").
+      hidden_size (RNN only): Size of the hidden layers (int, default = 256).
+      embed_size: Size of the embedding layer (int, default = 256).
+      dropout_rate: Dropout rate for the RNN output nodes (float, default = 0.0).
+      embed_dropout_rate: Dropout rate for the embedding layer (float, default = 0.0).
+
+#### Joint network
+
+Currently, we only propose the standard joint network module composed of three linear layers and an activation function. The module definition is optional but the following parameters can be modified through the configuration parameter `joint_network_conf`:
+
+    joint_network_conf:
+      joint_space_size: Size of the joint space (int, default = 256).
+      joint_act_type: Type of activation in the joint network (str, default = "tanh").
+
+The options related to the activation functions can also be modified through the parameters introduced in the Encoder section (See `main_conf` description).
+
+### Multi-task learning
+
+We also support multi-task learning with two auxiliary tasks: CTC and cross-entropy w/ label smoothing option (called LM loss here). The auxiliary tasks contribute to the overal task defined as:
+
+**L_tot = (λ_trans x L_trans) + (λ_auxCTC x L_auxCTC) + (λ_auxLM x L_auxLM)**
+
+where the losses (L_*) are respectively, in order: The Transducer loss, the CTC loss and the LM loss. Lambda values define their respective contribution to the total loss. Each task can be parameterized using the following options, passed to `model_conf`:
+
+    model_conf:
+      transducer_weight: Weight of the Transducer loss (float, default = 1.0)
+      auxiliary_ctc_weight: Weight of the CTC loss. (float, default = 0.0)
+      auxiliary_ctc_dropout_rate: Dropout rate for the CTC loss inputs. (float, default = 0.0)
+      auxiliary_lm_loss_weight: Weight of the LM loss. (float, default = 0.2)
+      auxiliary_lm_loss_smoothing: Smoothing rate for LM loss. If > 0, label smoothing is enabled. (float, default = 0.0)
+
+**Note:** We do not support other auxiliary tasks in ESPnet2 yet.
+
+### Inference
```

Comment:
In this section, the hyper-parameter names have hyphens `-` instead of underscores `_`. It would be better to make them consistent.